design/plugininputs-field.md

```diff
@@ -0,0 +1,115 @@
+    - [Use case 4:](#use-case-4)
+- [Detailed Design](#detailed-design)
+- [Alternatives Considered](#alternatives-considered)
+- [Security Considerations](#security-considerations)
+- [Compatibility](#compatibility)
+- [Open Issues](#open-issues)
+
+## Abstract
+This proposal aims to add a generic pluginInputs field in the velero CRs (Backup, Schedule, Restore) to allow the user to specify the parameters which are specific to a plugin. This will allow plugin owners to expose their own settings in the velero CRs. 
+
+## Background
+Velero plugins (backup item action, restore item action)  provide flexibility to extend velero functionality. However, the current design of the (backup item action, restore item action) plugins does not enable the user to specify the settings specific to a plugin in the backup/restore CR. Plugin owners have to take the route of taking inputs through annotations, configmap with fixed naming/label conventions for global settings. This is not an ideal end use experience since we are bloating annotations, having limitations w.r.t global settings for all backups/schedules etc.
+
+## Goals
+- Add pluginInputs field in the velero CRs (Backup, Schedule, Restore) to allow the user to specify the parameters which are specific to a plugin.
+
+# Precedence of this Design principle 
+This approach is similar to all current velero / K8s storage design principles.
+- **BackupStorageLocation** - we refer the name of the object store plugin provider(for eg "azure"), and below under config, we can specify the config specific to the plugin which is a generic property bag. Only the specific object store plugin can understand the config.
+- **CSI based VolumeSnapshotClass** - we refer the name of the CSI driver(for eg "csi.cloud.disk.driver"), and below under parameters, we can specify the parameters specific to the plugin which is a generic property bag. Parameters such as SKU etc are only understood by the specific CSI driver.
+- **VolumeSnapshotLocation** - we refer the name of the volume snapshotter plugin provider(for eg "aws"), and below under config, we can specify the config specific to the plugin which is a generic property bag. Properties such as region etc are only understood by the specific volume snapshotter plugin.
+
+
+### Use case examples for other plugins
+
+#### Use case 1: Change StorageClass RIA (velero.io/change-storage-class)
+Change Storage Class plugin, uses a global configmap which is discovered through labels/ annotations which is not ideal customer experience and prohibhit usage of specific settings for each backup/ schedule setup.
+
+The pluginInputs field can be used to pass in the reference the configmap/ secret to use for a particular plugin. This will allow the user to use different configmaps/ secrets for different backups/ schedules.
+
+Example: 
+```yaml
+spec:
+    pluginInputs:
+    - name: velero.io/storageclass
```

Comment:
Is the `name` field used to represent the plugin's name?
If so, the name should be `velero.io/change-storage-class`.